master/internal/rm/kubernetesrm/jobs.go

```diff
@@ -2212,6 +2220,103 @@ func (j *jobsService) removeEmptyNamespace(namespaceName string, clusterName str
 	return nil
 }
 
+func (j *jobsService) setResourceQuota(quota int, namespace string) error {
+	var k8sDeterminedLabel = map[string]string{determinedLabel: namespace}
+
+	k8sNamespace, err := j.clientSet.CoreV1().Namespaces().Get(context.TODO(), namespace,
+		metaV1.GetOptions{
+			TypeMeta: metaV1.TypeMeta{
+				Kind:       "Namespace",
+				APIVersion: "v1",
+			},
+		})
+	if err != nil {
+		return fmt.Errorf("error finding namespace %s: %w", namespace, err)
+	}
+
+	if _, ok := k8sNamespace.Labels[determinedLabel]; !ok {
+		return fmt.Errorf("cannot set quota on namespace %s. Namespace needs determined label",
+			namespace)
+	}
+
+	quotaName := namespace + "-quota"
+
+	// We want to patch the smallest quota if it is a determiend quota regardless of whether it's
+	// larger or smaller. If the quota does not correspond to the auto-generated namespace and is
+	// less than the current quota (the first non-Determined quota, we error out saying that they
+	// should remove that quota from Kubernetes and then try to set a Determined quota.
+	k8sQuotas, err := j.clientSet.CoreV1().ResourceQuotas(namespace).List(context.TODO(),
+		metaV1.ListOptions{})
+	quotas := k8sQuotas.Items
+	currentQuota := float64(quota)
+	var detQuota *k8sV1.ResourceQuota
+	for _, q := range quotas {
+		qResources := q.Spec.Hard
+		for name, quantity := range qResources {
+			if name == "requests."+ResourceTypeNvidia {
+				tmpQuota := quantity.AsApproximateFloat64()
```

Comment:
Why are `currentQuota` and `tmpQuota` cast to floats in this function if they're ultimately cast back to integers to create the quota resource and for error output?

Reply:
I am guessing it is for the most accurate comparison - but I am not sure since I didn't write the code. is this a blocker to merging the PR?

Reply:
Converting an integer to a float doesn't necessarily get you more accuracy; it can lead to subtle bugs, because of precision problems due to how IEEE-754 represents values. In this case, I believe it would be okay, because `float64` can represent all values between -2^53 and 2^53, and resource quotas fall well within that range, but outside of that range, floats will get rounded in specific and maybe unexpected ways. Performing arithmetic on those values can compound errors, too. If there's no reason to cast, we shouldn't cast. If the K8s API takes integer resource quotas, then the inputs can all remain integers.

None of these are blocking because I'm on the Infra team and not versed well enough in the day-to-day of this code, so I try not to hold up pull requests unless there's an infrastructure-specific thing that needs changing. But I'm still part of the review rotation, so I do my best to offer honest PR reviews and point out things I see.

Reply:
Makes sense, thanks for the detailed review! I will merge the PR as is for now since it is only merging into the feature branch and I will ask Amanda what her reasoning was, and we can write another PR if needed